rules_php_gapic/php_gapic.bzl

```diff
@@ -71,12 +71,14 @@ def php_gapic_srcjar(
         transport = "grpc+rest"
     if transport != "grpc+rest" and transport != "rest" and transport != "grpc":
         fail("Error: Only 'grpc+rest', 'rest' or `grpc` transports are supported")
+    if transport == "grpc" and migration_mode != "NEW_SURFACE_ONLY":
+        fail("Error: 'grpc' transport is only supported with 'NEW_SURFACE_ONLY' migration mode")
```

Comment:
this ensures that the previous client surfaces (e.g. `GapicClientGenerator`, as opposed to `GapicClientV2Generator`) are not able to be generated with "grpc-only" transport, for simplicity.